crates/ln-dlc-node/src/lib.rs

```diff
@@ -106,6 +107,9 @@ pub struct PaymentInfo {
     pub timestamp: OffsetDateTime,
     pub description: String,
     pub invoice: Option<String>,
+    /// If the payment was used to open an inbound channel, this tx id refers the funding
+    /// transaction for opening the channel.
+    pub funding_txid: Option<Txid>,
```

Comment:
🙃 Perhaps not worth it here, but it's nice to think about ways to avoid optional fields.

For instance, we could extend the `PaymentFlow` enum to something like:

```rust
enum PaymentType {
    Inbound,
    Outbound,
    Funding {
        funding_txid: Txid,
    }
}
```

Reply:
I like it 👍